adal/tests/Test.ADAL.NET.UIAutomation/CoreMobileADALTests.cs

```diff
@@ -37,37 +40,131 @@ namespace Test.ADAL.NET.UIAutomation
+        }
+
+        /// <summary>
+        /// Runs through the standard acquire token silent flow
+        /// </summary>
+        /// <param name="controller">The test framework that will execute the test interaction</param>
+        public static void AcquireTokenSilentTest(ITestController controller, UserQueryParameters userParams)
+        {
+            AcquireTokenInteractivly(controller, userParams);
+            VerifyResult(controller);
+
+            //Enter 2nd Resource
+            controller.EnterText(UiTestConstants.resourceEntryID, UiTestConstants.Exchange, false);
+            controller.DismissKeyboard();
+
+            //Acquire token silently
+            controller.Tap(UiTestConstants.AcquireTokenSilentID);
 
-            controller.Tap("secondPage");
+            VerifyResult(controller);
+        }
+
+        /// <summary>
+        /// Runs through the standard acquire token interactive flow
+        /// </summary>
+        /// <param name="controller">The test framework that will execute the test interaction</param>
+        public static void AcquireTokenADFSvXInteractiveTest(ITestController controller, bool isFederated, UserQueryParameters userParams)
+        {
+            AcquireTokenInteractivly(controller, userParams);
+            VerifyResult(controller);
+        }
+
+        private static void AcquireTokenInteractivly(ITestController controller, UserQueryParameters userParams)
+        {
+            var user = prepareForAuthentication(controller, userParams);
+            SetInputData(controller, UiTestConstants.UiAutomationTestClientId, UiTestConstants.MSGraph);
+            PerformSignInFlow(controller, user);
+        }
+
+        private static IUser prepareForAuthentication(ITestController controller, UserQueryParameters userParams)
+        {
+            //Navigate to second page
+            controller.Tap(UiTestConstants.secondPageID);
 
             //Clear Cache
-            controller.Tap("clearCache");
+            controller.Tap(UiTestConstants.ClearCacheID);
+
+            //Get User from Lab
+            return controller.GetUser(userParams);
+        }
+
+        private static void SetInputData(ITestController controller, string ClientID, string Resource)
+        {
+            //Enter ClientID
+            controller.EnterText(UiTestConstants.clientIdEntryID, ClientID, false);
+            controller.DismissKeyboard();
+
+            //Enter Resource
+            controller.EnterText(UiTestConstants.resourceEntryID, Resource, false);
+            controller.DismissKeyboard();
+        }
+
+        private static void PerformSignInFlow(ITestController controller, IUser user)
+        {
+            string passwordInputID = string.Empty;
+            string signInButtonID = string.Empty;
+
+            switch (user.FederationProvider)
+            {
+                case FederationProvider.AdfsV4:
+                    passwordInputID = UiTestConstants.AdfsV4WebPasswordID;
```

Comment:
I am a bit confused here - why do we need passwords ? Is this for the U/P flow? The class and the methods do not make this clear.

Reply:
This is meant to be more of a helper class for the automation tests. I have renamed it so that it makes more sense. This function is a generic sign in pattern for interactive flows on mobile platforms. Thus, it is not specific to any test.